packages/dd-trace/src/encode/0.4.js

```diff
@@ -263,6 +271,45 @@ class AgentEncoder {
     }
   }
 
+  _encodeObject (bytes, value, circularReferencesDetector = new Set()) {
+    circularReferencesDetector.add(value)
+    if (Array.isArray(value)) {
+      return this._encodeObjectAsArray(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'object') {
+      return this._encodeObjectAsMap(bytes, value, circularReferencesDetector)
+    } else if (typeof value === 'string' || typeof value === 'number') {
+      this._encodeValue(bytes, value)
+    }
+  }
+
+  _encodeObjectAsMap (bytes, value, circularReferencesDetector) {
+    const keys = Object.keys(value)
+    const validKeys = keys.filter(
+      key =>
+        typeof value[key] === 'string' || typeof value[key] === 'number' ||
+        (value[key] !== null && typeof value[key] === 'object' && !circularReferencesDetector.has(value[key])))
+
+    this._encodeMapPrefix(bytes, validKeys.length)
+
+    for (const key of validKeys) {
+      this._encodeString(bytes, key)
+      this._encodeObject(bytes, value[key], circularReferencesDetector)
+    }
+  }
+
+  _encodeObjectAsArray (bytes, value, circularReferencesDetector) {
+    const validValue = value.filter(
+      item =>
+        typeof item === 'string' || typeof item === 'number' ||
+        (item !== null && typeof item === 'object' && !circularReferencesDetector.has(item)))
+
+    this._encodeArrayPrefix(bytes, validValue)
```

Comment:
not sure how the function works but looking at `encodeMapPrefix()` it looks like this is missing a `.length` ?
```suggestion
    this._encodeArrayPrefix(bytes, validValue.length)
```

Reply:
Nope, the implementations are different and I haven't changed it. 
_encodeArrayPrefix:
https://github.com/DataDog/dd-trace-js/blob/52da415e4748e6cc821bff53d5f4e99b820ab4f7/packages/dd-trace/src/encode/0.4.js#L130-L135

__encodeMapPrefix:
https://github.com/DataDog/dd-trace-js/blob/52da415e4748e6cc821bff53d5f4e99b820ab4f7/packages/dd-trace/src/encode/0.4.js#L144-L154